examples/platform/telink/common/src/AppTaskCommon.cpp

```diff
@@ -157,51 +157,41 @@ class AppFabricTableDelegate : public FabricTable::Delegate
     {
         if (chip::Server::GetInstance().GetFabricTable().FabricCount() == 0)
         {
-            bool isBasicCommissioningMode = chip::Server::GetInstance().GetCommissioningWindowManager().GetCommissioningMode() ==
-                Dnssd::CommissioningMode::kEnabledBasic;
             ChipLogProgress(DeviceLayer, "Performing erasing of settings partition");
 
-#ifdef CONFIG_CHIP_FACTORY_RESET_ERASE_NVS
-            void * storage = nullptr;
-            int status     = settings_storage_get(&storage);
-
-            if (status == 0)
+            // Do FactoryReset in case of failed commissioning to allow new pairing via BLE
+            if (sIsCommissioningFailed)
```

Comment:
But if PASE did get established and then commissioning fails, won't this test false?

Approving, but it feels like this code is flailing around trying to implement some behavior but not being clear about what that behavior is and how to implement it...

Reply:
I've checked this case. The system works as we expect it to.

We clearly understand how this part should work and what test cases verify it (TC-CADMIN-1.19, TC-OPCREDS-3.6, checking the commissioning fail), but after the previous PR (https://github.com/project-chip/connectedhomeip/pull/28808) it turned out that the previous approach was not obvious, so we decided to update the part (use OnCommissioningSessionEstablishmentError) when we received a request to change CHIP_FACTORY_RESET_ERASE_NVS=n

Reply:
@bzbarsky-apple 
If you have no more questions, please resolve the conversation.

Reply:
It's your platform, in the end, so as long as this was reviewed by someone else at telink....